webapp/src/components/sidebar_right/github_items.jsx

```diff
@@ -7,10 +7,32 @@ import PropTypes from 'prop-types';
 import {Badge} from 'react-bootstrap';
 import {makeStyleFromTheme, changeOpacity} from 'mattermost-redux/utils/theme_utils';
 
+//retain previous item to show deleted items between 2 refresh call
+let previousItems = [];
```

Comment:
@manland, can we promote the `GithubItems` to a `React.Component` and store these as state? (Or, if you're adventurous, to a React Hook?)

While this likely works because we only render one `GitHubItems` at a time, it runs the risk of breaking in edge cases where React decides not to re-render the component, and it makes it hard to properly manipulate this state and achieve a re-render.

Reply:
I'm adventurous^^ please review again :sweat_smile: I keep this conversation open because tests are missing ;)